docs/advanced-features/sitemaps.md

```diff
@@ -0,0 +1,198 @@
+---
+description: Learn how to add a sitemap to your Next.js project, along with a robots.txt file for optimal SEO.
+---
+
+# Sitemaps
+
+<details open>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/with-sitemap">with-sitemap</a></li>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/with-next-sitemap">with-next-sitemap</a></li>
+  </ul>
+</details>
+
+To improve your Search Engine Optimization (SEO), you might want to add a [sitemap](https://developers.google.com/search/docs/advanced/sitemaps/build-sitemap) or [`robots.txt`](https://developers.google.com/search/docs/advanced/robots/intro) file to your Next.js website.
+
+A **sitemap** defines the relationship between pages of your site. Search engines utilize
+this file to more accurately index your site. You can also provide additional information
+such as last updated time, how frequently the page changes, and more.
+
+A **robots.txt** file tells search engines which pages or files the crawler can or can't request from your site.
+
+## Static Sitemaps
+
+If your website does not update frequently, you can manually create a static sitemap.
+This is a basic `.xml` file defining the content of your site. Here's a simple example:
+
+```xml
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+  <url>
+      <loc>https://nextjs.org</loc>
+  </url>
+  <url>
+      <loc>https://nextjs.org/docs</loc>
+  </url>
+</urlset>
+```
+
+## Dynamic Sitemaps
+
+If your site frequently changes, you should dynamically create a sitemap. First, let's explore an example where your content is based on [file-system routing](/docs/routing/introduction.md) (e.g., contained inside the `/pages` directory).
```

Comment:
> example where your content is based on [file-system routing](/docs/routing/introduction.md)

if i am reading this correctly, the example does not handle dynamic route segments, correct?